modules/ding_campaign/ding_campaign.bpi.inc

```
@@ -0,0 +1,108 @@
+
+    return array(
+      'bpi_field_title' => array(
+        '#type' => 'select',
+        '#title' => t('BPI title'),
+        '#description' => t('Titles are automatically assigned.'),
+        '#options' => array('title' => t('Title')),
+        '#value' => 'title',
+        '#disabled' => TRUE,
+      ),
+
+      'bpi_field_body' => array(
+        '#type' => 'select',
+        '#title' => t('BPI body'),
+        '#description' => t('Field to extract the main content from (body field).'),
+        '#options' => $field_options,
+        '#default_value' => isset($field_mapping['bpi_field_body']) ? $field_mapping['bpi_field_body'] : NULL,
+      ),
+
+      'bpi_field_url' => array(
+        '#type' => 'select',
+        '#title' => t('BPI url'),
+        '#description' => t('Field to extract the url from.'),
+        '#options' => $field_options,
+        '#default_value' => isset($field_mapping['bpi_field_url']) ? $field_mapping['bpi_field_url'] : NULL,
+      ),
+    );
+  }
+
+  return NULL;
+}
+
+/**
+ * Implements hook_bpi_convert_to_bpi_alter().
+ */
+function ding_campaign_bpi_convert_to_bpi_alter(&$bpi_content, $node, $mapping) {
+  if (isset($mapping['bpi_node_type']) && in_array($mapping['bpi_node_type'], ding_campaign_bpi_syndicate_get_bpi_types())) {
+    $url_field = isset($mapping['bpi_field_mapping']['bpi_field_url']) ? field_view_field('node', $node, $mapping['bpi_field_mapping']['bpi_field_url']) : NULL;
+    $url = bpi_get_field_value($url_field);
+    $bpi_content['url'] = $url;
+
+    $campaign_type = bpi_get_field_value(field_view_field('node', $node, 'field_camp_settings'));
+    $bpi_content['data'] = array(
+      'type' => $campaign_type,
+    );
+  }
+}
+
+/**
+ * Implements hook_bpi_syndicate_form().
+ */
+function ding_campaign_bpi_syndicate_form_alter(&$form, $bpi_content, $syndicated_images) {
+  $current_language = $form['language']['#value'];
+  $data = @json_decode($bpi_content['data'], TRUE);
```

Why do we want to suppress errors here? That is generally not a good practice.

---

You're absolutely right. I don't know why I did that – will fix!